packages/client/hmi-client/src/components/responsive-matrix/label-cols.vue

```diff
@@ -0,0 +1,173 @@
+<template>
+	<div class="label-container col" :style="labelStyle">
+		<div class="label" v-for="(value, idx) in labelValues" :key="idx" :style="getLabelStyle(idx)">
+			{{ value }}
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import { PropType } from 'vue';
+
+import { Point } from 'pixi.js';
+
+import { CellStatus } from '@/types/ResponsiveMatrix';
+
+// EMPTY_POINT used as a fallback value
+const EMPTY_POINT = new Point();
+
+export default {
+	// ---------------------------------------------------------------------------- //
+	// props                                                                        //
+	// ---------------------------------------------------------------------------- //
+
+	props: {
+		viewport: {
+			type: null,
+			default() {
+				return null;
+			}
+		},
```

Comment:
It is easier in my opinion - to make viewport required, and add a guard at the calling layer - since it doesn't make any sense to render the labels without a viewport anyway, and cleans up a bunch of question-marks.

```
<blahblah>
   <label-cols v-if="viewport && something" ... />
</blahblah>
```

Reply:
i like this idea but this doesnt work unfortunately, because viewport is not reactive (due to reactivity breaking the component). therefore when mounted viewport will be false and it will not update automatically.